src/test/regress/expected/shard_rebalancer.out

```diff
@@ -67,6 +67,35 @@ SELECT count(*) FROM citus_local_table;
      1
 (1 row)
 
+-- verify drain_node uses the localhostname guc by seeing it fail to connect to a non-existing name
+ALTER SYSTEM SET citus.local_hostname TO 'foobar';
+SELECT pg_reload_conf();
+ pg_reload_conf
+---------------------------------------------------------------------
+ t
+(1 row)
+
+SELECT pg_sleep(1); -- wait to make sure the config has changed before running the GUC
```

Comment:
Since the hostname is only used in the current backend, can we just do `SET citus.local_hostname TO 'foobar'`?

Reply:
The setting requires an ALTER SYSTEM SET  currently, and as per discussion above should have been a super user setting. The setting is not intended to be changable by normal users/sessions as it will trick the coordinator in connecting somewhere you might not want it to connect to for certain operations.

I don't fully understand why the setting was not directly reflected to the new value after the reload, however, I feel it has todo with the signal to postmaster which subsequently signals all the backends to reload and then waiting for a safe point to reload which shows some racy behaviour both locally and on CI.

Reply:
In most other places we do this we use `pg_sleep(0.1)` I don't think we've had problems where 100ms sleep wasn't enough. Can you use that here too?

Reply:
> I don't fully understand why the setting was not directly reflected to the new value after the reload, however, I feel it has todo with the signal to postmaster which subsequently signals all the backends to reload and then waiting for a safe point to reload which shows some racy behaviour both locally and on CI.

I have seen this in the regression tests as well for other GUCs, and concluded similarly

Reply:
I think the main reason for this sleep to be needed here (and in various other tests), is that all `pg_reload_conf` does is send a `SIGHUP` to postmaster. Since signal handling is asynchronous, the setting isn't actually applied at the point when `pg_reload_conf` returns.

Reply:
I have reduced the wait time. I think there are still some sleep(1)'s in the testsuite. Didn't want to change them all but might be a nice thing todo after the freeze to optimize our test suite.